bundle-workflow/python/git/git_repository.py

```diff
@@ -19,9 +19,7 @@ def __init__(self, url, ref, directory = None):
             self.dir = self.temp_dir.name
         else:
             self.dir = directory
-            print(f'Deleting {self.dir}')
```

Comment:
I think there was utility in being sure the folder was cleaned up? Maybe we add a default that causes this to barf unless you've set `allow-nuke=true`

Reply:
We checkout into a temporary folder, so these actually do get cleaned up. Making GitRepository act like a temporary folder itself seems unnecessary, but I would be fine if we needed that. The problem here is that this class is cleaning up something it didn't create!